src/concern/mod.rs

```diff
@@ -244,18 +246,9 @@ impl WriteConcern {
         self.w != Some(Acknowledgment::Nodes(0)) || self.journal == Some(true)
     }
 
-    /// Validates that the write concern. A write concern is invalid if the `w` field is 0
+    /// Validates that the write concern. A write concern is invalid if both the `w` field is 0
     /// and the `j` field is `true`.
-    pub fn validate(&self) -> Result<()> {
-        if let Some(Acknowledgment::Nodes(i)) = self.w {
-            if i < 0 {
-                return Err(ErrorKind::ArgumentError {
-                    message: "write concern `w` field cannot be negative integer".to_string(),
-                }
-                .into());
-            }
-        }
-
+    pub(crate) fn validate(&self) -> Result<()> {
```

Comment:
Why was this removed from the public API?

Reply:
Oh good catch, I meant to call this out in a comment but forgot to. This method seemed really niche and like something that was more useful for us internally than for users, especially since it requires an unacknowledged write concern which we don't currently support.

The method originally stems from this requirement in the [spec](https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#inconsistent-writeconcern):
>Drivers MUST raise an error when an inconsistent WriteConcern is specified.

From my reading, this doesn't require that users need to be able to eagerly determine if a WC is valid or even that an error needs to be thrown when an inconsistent WC is created, just when one is "specified". So as long as we throw an error operation execution, we should still be abiding by the spec in my eyes. I doubt users were using this method anyways. Also, we throw errors on all `w: 0` write concerns, so its a bit of a moot point anyways.

I filed RUST-784 to cover this change so that we remember to include it in the release notes as a breaking change.

Reply:
Gotcha, yeah I agree with your reading of the spec. Even if we were to allow unacknowledged writes I would guess most users would rely upon us to validate their write concerns rather than doing it themselves, so this seems fine to me to remove.